File: Java/Longest_Common_Prefix.java
class Solution {

    private boolean charNotEqual(String[] strs, int i, int j) {
        if(j >= strs[i].length() || j >= strs[i+1].length()) return true;
        return strs[i].charAt(j) != strs[i+1].charAt(j);
    }
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0) return "";
        for (int j = 0; j < strs[0].length(); j++) {
            for (int i = 0; i < strs.length - 1; i++) {
                if(charNotEqual(strs, i, j)) return strs[0].substring(0, j);
            }
        }
        return strs[0];
    }
}

File: CPP/Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==0)
        {
            return"";
        }
        string ans="";
        for(int i=0; i<strs[0].size(); i++)
        { bool s=true;
            for(int j=1; j<strs.size(); j++)
            {
                if(strs[0][i]!=strs[j][i])
                {
                    s=false;
                }
            }
         if(s==true)
         {
             ans+=strs[0][i];
         }
         else{
             return ans;
         }
        } return ans;
    }
};
